src/shell/output.c

```diff
@@ -586,37 +586,84 @@ static int shell_output_parse_type (struct shell_output *out,
     return 0;
 }
 
-/* handle mustache templates and the similar special cases */
-static char *shell_output_get_path (struct shell_output *out, const char *path)
-{
-    char *rv = NULL;
-    char *ptr;
-
-    /* replace {{id}} with jobid */
-    if ((ptr = strstr (path, "{{id}}"))) {
-        char buf[32];
-        int len, buflen, total_len;
-
-        len = strlen (path);
-        buflen = snprintf (buf, sizeof (buf), "%ju",
-                           (uintmax_t)out->shell->info->jobid);
-        /* -6 for {{id}}, +1 for NUL */
-        total_len = len - 6 + buflen + 1;
-        if (total_len > (PATH_MAX + 1)) {
-            errno = EOVERFLOW;
-            return NULL;
+static int fwrite_shell_info (flux_shell_t *shell, FILE *fp, const char *name)
+{
+    char value[128];
+
+    /*  "jobid" is a synonym for "id" */
+    if (strncmp (name, "jobid", 5) == 0)
+        name += 3;
+    if (strncmp (name, "id", 2) == 0) {
+        const char *type = "f58";
+        if (strlen (name) > 2) {
+            if (name[2] != '.') {
+                shell_log_error ("bad template name %s", name);
+                return -1;
+            }
+            type = name+3;
+        }
+        if (flux_job_id_encode (shell->info->jobid,
+                                type,
+                                value,
+                                sizeof (value)) < 0) {
+            shell_log_errno ("flux_job_id_encode failed for %s", name);
+            return -1;
         }
-        if (!(rv = calloc (1, total_len)))
-            return NULL;
-        memcpy (rv, path, ptr - path);
-        memcpy (rv + (ptr - path), buf, buflen);
-        memcpy (rv + (ptr - path) + buflen, ptr + 6, len - (ptr - path) - 6);
     }
-    else {
-        if (!(rv = strdup (path)))
-            return NULL;
+    return fputs (value, fp);
```

Comment:
i think if `name` is not `jobid` or `id`, I think you'll be writing the uninitialized `value` to `fp`?  I'm guessing this is why I saw this when playing around:

```
>flux mini run --output="job-{{foobar}}.out" hostname
0: stdout redirected to job-5.out
0: stderr redirected to job-5.out
```